spec/cohort.spec.js

```diff
@@ -0,0 +1,94 @@
+import Cohort from '../src/cohort.js'
+
+describe('Cohort:', () => {
+  beforeEach(() => {
+    Cohort.resetId()
+  })
+
+  describe('new instance -', () => {
+    it('valid name input & capacity input', () => {
+      const cohort = new Cohort('Potato', 16)
+
+      const expected = {
+        name: 'Potato',
+        cohortId: 1,
+        capacity: 16
+      }
+
+      expect(cohort.details).toEqual(expected)
+    })
+
+    it('valid name input & no capacity input', () => {
+      const cohort = new Cohort('Potato')
+
+      const expected = {
+        name: 'Potato',
+        cohortId: 1,
+        capacity: 24
+      }
+
+      expect(cohort.details).toEqual(expected)
+    })
+
+    it('valid name input & capacity input not Number', () => {
+      const inputs = ['seven', '7', { 1: 1 }, [1]]
+
+      inputs.forEach((input) => {
+        const callback = () => new Cohort('Potato', input)
+
+        expect(callback).toThrowError('capacity input must be Number')
+      })
+    })
+
+    it('name input not String', () => {
+      const inputs = [7, { seven: 'seven' }, ['seven']]
+
+      inputs.forEach((input) => {
+        const callback = () => new Cohort(input, 16)
+
+        expect(callback).toThrowError('name input must be String')
+      })
+    })
+
+    it('name input length must be greater than 0', () => {
+      const callback = () => new Cohort('', 16)
+
+      expect(callback).toThrowError('name input must not be blank')

```

Comment:
Curious what you think about extracting the error message out into a variable as well? 

const expectation = 'name input must not be blank'
const callback = () => new Cohort('', 16)

expect(callback).toThrowError(expectation)

Reply:
If you want to go all out, you could define an error message object which contains every single error message for your app then use that in both the source code and the test suite

Reply:
Interesting, makes sense! Something like the below is what you had in mind @vherus?
```js
const errorMessages = {
  inputStringReq = `input must be of type: String`
  ...
}
```

Reply:
That's right, then you have a single source of truth for all of the error messages your app can throw. Much easier to maintain, more consistent, and gives you room to evolve your error handling system over time